packages/react-scripts/config/webpackDevServer.config.js

```diff
@@ -60,4 +62,10 @@ module.exports = {
   https: protocol === 'https',
   host: host,
   overlay: false,
+  historyApiFallback: {
+    // Paths with dots should still use the history fallback.
+    // See https://github.com/facebookincubator/create-react-app/issues/387.
+    disableDotRule: true,
+  },
```

Comment:
Thanks for reporting this, I'll have to investigate to see what's happening. `disableDotRule` is used in the original proxy implementation and there's obviously a conflict when using the proxy object.